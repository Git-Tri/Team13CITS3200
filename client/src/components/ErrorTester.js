import React from "react";

/**
 * Catches an error and uses the onError callback to allow tests to pass
 * this component is meant for testing purposes only
 * when testing place the error component as a child to this component 
 * then pass in an onError prop or check the state of this component
 * derived from: https://codepen.io/gaearon/pen/wqvxGa?editors=0010
 */
class ErrorTester extends React.Component {
    constructor(props) {
      super(props);
      this.state = { error: null, errorInfo: null };
    }
    
    componentDidCatch(error, errorInfo) {
      this.setState({
        error: error,
        errorInfo: errorInfo
      })

      this.props.onError(error,errorInfo);
    }
    
    render() {
      if (this.state.errorInfo) {
        return (
          <p> Good news an error occurred</p>
          );
      }
      return this.props.children;
    }  
  }

  export default ErrorTester;
