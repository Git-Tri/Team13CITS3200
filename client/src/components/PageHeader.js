import React, { Component } from 'react';
import { Header, Button, Grid } from 'semantic-ui-react';

class PageHeader extends Component {


	render() {
		return (
			<Grid container columns={3}>
				<Grid.Row>
					<Grid.Column>
						<Button onClick={this.props.handleSidebarClick} >
							{(this.props.sidebarVisible?"Hide":"Show")+" sidebar"}
						</Button>
					</Grid.Column>
					<Grid.Column>
					<Header>
						{this.props.header}
					</Header>
					</Grid.Column>
					<Grid.Column>
					<Button onClick={() => {window.history.back()}}>
						Back
					</Button>
					</Grid.Column>
				</Grid.Row>
			</Grid>
		);
	}
}

export default PageHeader;