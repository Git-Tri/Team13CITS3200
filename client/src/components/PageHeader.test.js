import PageHeader from "./PageHeader";
import renderer from 'react-test-renderer';
import React from 'react';
import ErrorTester from "./ErrorTester";
import { shallow, mount } from "enzyme";
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe("PageHeader Tests",function()
{   
    test("should render correctly",() => 
    {
        var mockSidebarClick = jest.fn()
        let wrapper = shallow(<PageHeader
                                header={"Match List"}
                                sidebarVisible={true}
                                handleSidebarClick={mockSidebarClick}
                            />)
        expect(wrapper).toMatchSnapshot();
    })
})
