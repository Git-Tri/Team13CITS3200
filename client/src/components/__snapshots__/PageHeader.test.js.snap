// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader Tests should render correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`PageHeader Tests should render correctly with content 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`PageHeader Tests should render correctly with isbackable 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
